cmd/acra-connector/acra-connector.go

```diff
@@ -186,37 +195,69 @@ func main() {
+
+	log.Infof("Preparing to start in mode: %s", connectorMode)
+
+	outgoingConnectionString := ""
+	outgoingSecureSessionId := ""
+
+	// if AcraTranslator
+	if connectorMode == connector_mode.AcraTranslatorMode {
+		if *acraTranslatorHost == "" && *acraTranslatorConnectionString == "" {
+			log.WithField(logging.FieldKeyEventCode, logging.EventCodeErrorWrongConfiguration).
+				Errorln("Configuration error: you must pass acratranslator_connection_host or acratranslator_connection_string parameter")
+			os.Exit(1)
+		}
+		if *acraTranslatorPort != cmd.DEFAULT_ACRATRANSLATOR_GRPC_PORT {
+			*acraTranslatorConnectionString = network.BuildConnectionString(cmd.DEFAULT_ACRACONNECTOR_CONNECTION_PROTOCOL, *acraTranslatorHost, *acraTranslatorPort, "")
+		}
+		outgoingConnectionString = *acraTranslatorConnectionString
+		outgoingSecureSessionId = *acraTranslatorId
 	}
-	if *enableHTTPApi {
-		if *acraServerHost == "" && *acraServerApiConnectionString == "" {
+
+	// if AcraServer
+	if connectorMode == connector_mode.AcraServerMode {
+		if *acraConnectorPort != cmd.DEFAULT_ACRACONNECTOR_PORT {
+			*connectionString = network.BuildConnectionString(cmd.DEFAULT_ACRACONNECTOR_CONNECTION_PROTOCOL, cmd.DEFAULT_ACRACONNECTOR_HOST, *acraConnectorPort, "")
+		}
+		if *acraConnectorApiPort != cmd.DEFAULT_ACRACONNECTOR_API_PORT {
+			*connectionAPIString = network.BuildConnectionString(cmd.DEFAULT_ACRACONNECTOR_CONNECTION_PROTOCOL, cmd.DEFAULT_ACRACONNECTOR_HOST, *acraConnectorApiPort, "")
+		}
+
+		if *acraServerHost == "" && *acraServerConnectionString == "" {
 			log.WithField(logging.FieldKeyEventCode, logging.EventCodeErrorWrongConfiguration).
-				Errorln("Configuration error: you must pass acraserver_connection_host or acra_api_connection_string parameter")
+				Errorln("Configuration error: you must pass acraserver_connection_host or acraserver_connection_string parameter")
 			os.Exit(1)
 		}
 		if *acraServerHost != "" {
-			*acraServerApiConnectionString = network.BuildConnectionString(cmd.DEFAULT_ACRA_CONNECTION_PROTOCOL, *acraServerHost, *acraServerApiPort, "")
+			*acraServerConnectionString = network.BuildConnectionString(cmd.DEFAULT_ACRA_CONNECTION_PROTOCOL, *acraServerHost, *acraServerPort, "")
 		}
+		if *acraServerEnableHTTPApi {
+			if *acraServerHost == "" && *acraServerApiConnectionString == "" {
+				log.WithField(logging.FieldKeyEventCode, logging.EventCodeErrorWrongConfiguration).
+					Errorln("Configuration error: you must pass acraserver_connection_host or acra_api_connection_string parameter")
+				os.Exit(1)
+			}
+			if *acraServerHost != "" {
+				*acraServerApiConnectionString = network.BuildConnectionString(cmd.DEFAULT_ACRA_CONNECTION_PROTOCOL, *acraServerHost, *acraServerApiPort, "")
+			}
+		}
+
+		outgoingConnectionString = *acraServerConnectionString
+		outgoingSecureSessionId = *acraServerId
 	}
 
+	// --------- check client id and client key  -----------
```

Comment:
why hyphens?)
in a future to edit such comments we will need to move cursor through hyphens and only then update comment
or delete pre/post hyphens and edit text. instead just edit comment text)
Anyway, I think any developer will understand what we do when calling `ValidateClientId` and will not need any comment to understand :)